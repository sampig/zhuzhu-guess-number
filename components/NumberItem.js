import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

const styles = StyleSheet.create({
  container: {
    margin: 4,
    padding: 4,
    borderWidth: 2,
    borderRadius: 10,
    alignSelf: 'stretch',
    backgroundColor: 'rgba(255, 255, 255, 0.8)',
    flexDirection: 'row',
    alignItems: 'center',
  },
  itemSection: {
    marginLeft: 8,
    marginRight: 8,
    padding: 4,
    flex: 1,
    flexDirection: 'row',
  },
  idView: {
    flex: 1,
  },
  idText: {
    fontSize: 20,
  },
  guessView: {
    paddingLeft: 8,
    paddingRight: 8,
    flex: 1,
  },
  guessText: {
    fontSize: 20,
  },
  resultView: {
    paddingLeft: 8,
    paddingRight: 8,
    flexDirection: 'row',
    flex: 1,
  },
  correctCount: {
    paddingRight: 8,
    fontSize: 20,
    color: 'green',
  },
  incorrectCount: {
    fontSize: 20,
    color: 'red',
  },
})

/**
 * Component to display one guess result.
 * @param {object} numberItem - a guess object.
 * @returns {node}            - React node
 * @constructor
 */
function NumberItem ({ numberItem }) {

  /**
   * Title section.
   * @returns {node} - React node of title
   */
  const titleSection = () => {
    return (
      <View style={styles.itemSection}>
        <View style={styles.idView}>
          <Text style={styles.idText}>{numberItem.id}</Text>
        </View>
        <View style={styles.guessView}>
          <Text style={styles.guessText}>{numberItem.guess}</Text>
        </View>
        <View style={styles.resultView}>
          <Text style={styles.correctCount}>{numberItem.result[0] + 'âœ“'}</Text>
          <Text style={styles.incorrectCount}>{numberItem.result[1] + 'X'}</Text>
        </View>
      </View>
    )
  }

  return (
    <View style={styles.container}>
      {titleSection()}
    </View>
  )
}

export default NumberItem
