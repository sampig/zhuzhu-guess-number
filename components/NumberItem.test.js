import React from 'react'
import renderer from 'react-test-renderer'
import NumberItem from './NumberItem'

const item = { id: 'testId', guess: [1, 2, 3, 4], result: [2, 2] }

describe('The NumberItem component', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<NumberItem numberItem={item} />)
    const render = component.toJSON()
    expect(render).toMatchSnapshot()
  })
})
