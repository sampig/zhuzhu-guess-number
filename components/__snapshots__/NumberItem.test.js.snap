// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NumberItem component renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "rgba(255, 255, 255, 0.8)",
      "borderRadius": 10,
      "borderWidth": 2,
      "flexDirection": "row",
      "margin": 4,
      "padding": 4,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginLeft": 8,
        "marginRight": 8,
        "padding": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        testId
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        1
        2
        3
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "green",
            "fontSize": 20,
            "paddingRight": 8,
          }
        }
      >
        2âœ“
      </Text>
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 20,
          }
        }
      >
        2X
      </Text>
    </View>
  </View>
</View>
`;
